### DISCUSSION TITLE
Throttling of yfinance at certain hours

### DISCUSSION THREAD

**Discussion by AndrewBworth** [ranaroussi/yfinance#1678] [Q&A]
Hello, I've been using yfinance for a few weeks.  It seems that between 8pm and about 10:45 pm eastern time (0:00 to 2:45 UTC) the data I get from yfinance eventually shifts to data that has had too much information missing to repair.  Obviously after a certain number of queries, it just starts sending junk.  I'm assuming this is because this is prime time, possibly because candlestick patterns from yesterday are finalized.  I've been using repair="silent" which fixes the problems at other times in the day.  I've been trying to incorporate delays to fix the problem, but it doesn't work.  Is there a way around this problem, besides the obvious waiting three hours?

**Comment by ValueRaider** [comment]
Data problems around midnight? That's familiar https://github.com/ranaroussi/yfinance/discussions/1669

**Reply by AndrewBworth** [reply]
Yes, that's me.  Repair works on most of the data problems Yahoo produces, but after about 30 inquiries I start to get data problems that repair can't fix.  I realized I may not be out of ideas, I could put my inquiries into a more nested loop possibly, but I'm not sure if it will work.  I've seen somewhere supposedly there is a posted solution on someone's FAQ page, but I didn't understand it at all.

**Reply by ValueRaider** [reply]
> but after about 30 inquiries ...

Are you using a rate limiter? Best way to introduce delays, considering that repair can send out a few more requests for each original request.

**Reply by AndrewBworth** [reply]
I'm using a rate limiter, but it's of my own design so it doesn't work.  If Yahoo sends me data of size zero, a second history query seems to fix it.  If I'm asking for data of say 60 days, and it sends me back 59 days (one whole day missing), repair can't seem to fix it, and repeating the history query doesn't change anything.  I haven't yet tried nesting my ticker and history query into one more function, hoping to end the session and restart it completely.  I tried implementing a rate limiter that I made, if the data comes back size I wanted minus one, it waits ten seconds, and if it repeats the problem, it implements a delay ten seconds longer than the previous.  It doesn't work.  I've seen another solution somewhere, but I just didn't understand it
